rust/src/data_catalog/storage/mod.rs

```diff
@@ -0,0 +1,180 @@
+//! listing_schema contains a SchemaProvider that scans ObjectStores for tables automatically
+use std::any::Any;
+use std::collections::{HashMap, HashSet};
+use std::path::Path;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use dashmap::DashMap;
+use datafusion::catalog::schema::SchemaProvider;
+use datafusion::datasource::TableProvider;
+use datafusion_common::DataFusionError;
+use futures::TryStreamExt;
+use itertools::Itertools;
+use object_store::ObjectStore;
+
+use crate::storage::config::{configure_store, StorageOptions};
+use crate::{ensure_table_uri, open_table_with_storage_options, DeltaResult};
+
+const DELTA_LOG_FOLDER: &str = "_delta_log";
+
+/// A `SchemaProvider` that scans an `ObjectStore` to automatically discover delta tables.
+///
+/// A subfolder relationship is assumed, i.e. given:
+/// authority = s3://host.example.com:3000
+/// path = /data/tpch
+///
+/// A table called "customer" will be registered for the folder:
+/// s3://host.example.com:3000/data/tpch/customer
+///
+/// assuming it contains valid deltalake data, i.e a `_delta_log` folder:
+/// s3://host.example.com:3000/data/tpch/customer/_delta_log/
+pub struct ListingSchemaProvider {
+    authority: String,
+    /// Underlying object store
+    store: Arc<dyn ObjectStore>,
+    /// A map of table names to a fully quilfied storage location
+    tables: DashMap<String, String>,
+    /// Options used to create underlying object stores
+    storage_options: StorageOptions,
+}
+
+impl ListingSchemaProvider {
+    /// Create a new [`ListingSchemaProvider`]
+    pub fn try_new(
+        root_uri: impl AsRef<str>,
+        storage_options: Option<HashMap<String, String>>,
+    ) -> DeltaResult<Self> {
+        let uri = ensure_table_uri(root_uri)?;
+        let authority = uri.to_string();
```

Comment:
Why is `uri` turned into a String and then parsed again to be passed into `configure_store`? Can `authority` be removed here and the function just pass along `uri` for configuring the store?

Reply:
my best defence is that this is an artifact from an earlier code version 😆.